warehouse/accounts/services.py

```diff
@@ -226,6 +250,30 @@ def is_disabled(self, user_id):
         else:
             return (True, user.disabled_for)
 
+    def has_two_factor(self, user_id):
+        """
+        Returns True if the user has any form of two factor
+        authentication.
+        """
+        two_factor = self.get_two_factor(user_id)
+
+        # TODO: This is where user.u2f_provisioned et al.
+        # will also go.
+        return two_factor.totp_secret is not None
+
+    def check_totp_value(self, user_id, totp_value):
+        """
+        Returns True if the given TOTP is valid against the user's secret.
+
+        If the user doesn't have a TOTP secret, returns False.
+        """
+        two_factor = self.get_two_factor(user_id)
```

Comment:
Again with the ``two_factor`` can be ``None`` thing. Might make sense to just fix the ``get_two_factor`` function to raise an error so these can stay more focused.

Reply:
Ah, I guess this answers my question in https://github.com/pypa/warehouse/pull/5567#discussion_r273292255 😄. I'll make `get_two_factor` raise an error.